backend/bittan/bittan/services/swish/__init__.py

```diff
@@ -0,0 +1,163 @@
+			send_callback = True
+
+		model.error_code = payment_request_response.error_code
+		model.status = payment_request_response.status
+
+		model.swish_api_response = response_raw
+		model.save()
+
+		if send_callback:
+			payment_request = SwishPaymentRequest(model)
+			self.callback_function(payment_request)
+
+		return model
+
+	def synchronize_all_pending(self):
+		pending_payments = SwishPaymentRequestModel.objects.filter(status=SwishApiPaymentStatus.CREATED)
+
+		for pending in pending_payments:
+			self.synchronize_payment_request(pending)
+
+	def synchronize_payment_request(self, payment_request: SwishPaymentRequestModel | str):
+		""" This method is useful for when the callbacks are not working. It fetches the state of a payment via the swish api and updates our local info on the payment """
+
+		payment_request_id = None
+		if isinstance(payment_request, str):
+			payment_request_id = payment_request 
+			payment_request = None
+		else:
+			payment_request_id = payment_request.id
+
+		response = self.send_to_swish('GET', f'api/v1/paymentrequests/{payment_request_id}')
+		if response.status_code != 200:
+			# TODO Handle errors more elegantly, this should NOT happen!
+			logging.error("PaymentRequestDoes not exist:", payment_request_id)
+
+			raise Exception("There is no swish payment request with the id ", swish_payment_request)
+		
+		response_body = response.json()
+		return self.update_swish_payment_request(response_body, payment_request)
+
+	def get_payment_request(self, id: str) -> SwishPaymentRequest:
+		payment_request = SwishPaymentRequestModel.objects.get(pk=id)
+
+		# TODO check that payment requests only can change state once	
+		# If the payment status is created, there is a chance that it's status has been changed
+		if payment_request.status == SwishApiPaymentStatus.CREATED:
+			self.synchronize_payment_request(payment_request)
+
+		return SwishPaymentRequest(payment_request)
+
+
+	@staticmethod
+	def generate_swish_id() -> str: 
+		""" We have to provide an id when creating swish payment requests, this method generates valid id:s """
+		return str(uuid4()).replace('-', '').upper()
+	
+	def handle_swish_callback(self, response):
+		self.update_swish_payment_request(response)
+
+	def send_to_swish(self, method, path: str, **kwargs):
+		""" Convenience method for sending an HTTP request to the SWISH api """
+		return requests.request(method, f'{self.swish_url}{path}', cert=self.cert_file_paths, **kwargs)
+
+
+	def create_swish_payment(self, amount: int, message="") -> SwishPaymentRequest:
+			"""
+			Tells Swish that we want to create a payment. 
+
+			@param amount: The amount of the payment.
+			@param message: An optional message for the payment.
+			@return A representation of the paymentRequest
+			"""
+			payment_id = Swish.generate_swish_id()
+			
+
+			payment_request_db_object = SwishPaymentRequestModel(id=payment_id, amount=amount)
+			payment_request_db_object.save()
+
+			try:
```

Comment:
Can some lines be moved outside the try block?

Reply:
58ef2da7bf70bbc7b60dd576e464f8d87c8a7169